hazelcast/src/main/java/com/hazelcast/internal/nio/InstanceCreationUtil.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.nio;
+
+import org.objenesis.Objenesis;
+import org.objenesis.ObjenesisStd;
+import org.objenesis.instantiator.ObjectInstantiator;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
+
+import static com.hazelcast.internal.util.ExceptionUtil.rethrow;
+
+/**
+ * Utility class to deal with creation of new object instances.
+ */
+public final class InstanceCreationUtil {
+
+    private static final Objenesis OBJENESIS = new ObjenesisStd();
```

Comment:
Why did you not use [this](http://objenesis.org/apidocs/org/objenesis/ObjenesisStd.html#ObjenesisStd-boolean-) constructor which does do the caching for the underlying instantiator (no need for `OBJECT_SUPPLIERS` in that case)?